framework/src/org/checkerframework/framework/util/AtmLubVisitor.java

```diff
@@ -116,8 +116,12 @@ private void lubPrimaryAnnotations(
             AnnotatedTypeMirror type1, AnnotatedTypeMirror type2, AnnotatedTypeMirror lub) {
         Set<? extends AnnotationMirror> lubSet;
         if (visitingUninferedWildcard
-                        && type1.getAnnotations().size() != type1.getAnnotations().size()
-                || type1.getAnnotations().isEmpty()) {
+                && (type1.getAnnotations().size() != type2.getAnnotations().size()
```

Comment:
Thanks for the additional test!
`visitingUninferedWildcard` can only be true if at least one of the two annotation sets is empty, right?
So you can remove the partialQualifierLub method, see which one of the two is empty and then use the other one as LUB. I've updated #1235 to check whether either side is empty and expanded the test case to cover both cases. The test passed locally.